src/Event/Subscription/EventSubscriptionException.php

```diff
@@ -0,0 +1,23 @@
+<?php declare(strict_types=1);
+
+namespace Comquer\Event\Subscription;
+
+use RuntimeException;
+
+class EventSubscriptionException extends RuntimeException
+{
+    public static function unexpectedSubscriptionType(string $subscriptionType) : self
+    {
+        return new self("Unexpected subscription type: `$subscriptionType`");
+    }
+
+    public static function missingKeyFromArrayConfig(string $key) : self
+    {
+        return new self("Array config is missing a required key: `$key`");
```

Comment:
could be made a bit more explcit, consider clarifying where the key should be. Potentially just by referring to it as a top level key or something. Given the strict nature of the config schema, it seems like it would be helpful to give more of a hint of what that schema should be.

Reply:
Now that I think about it, @tjweldon, I don't see how we could pass any more information there that we already are. The missing key is there (it's the top level key, there isn't anything above that.) Information about what's throwing the exception and at which point of execution is in the stack (`EventSubscriptionCollection::fromConfig()`).